aws/resource_aws_elasticsearch_domain.go

```diff
@@ -713,8 +713,14 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 	// because DescribeElasticsearchDomainConfig does not return MasterUserOptions
 	if ds.AdvancedSecurityOptions != nil {
 		if _, ok := d.GetOk("advanced_security_options"); ok {
-			d.Set("advanced_security_options.0.enabled", ds.AdvancedSecurityOptions.Enabled)
-			d.Set("advanced_security_options.0.internal_user_database_enabled", ds.AdvancedSecurityOptions.InternalUserDatabaseEnabled)
+			if err := d.Set("advanced_security_options", []interface{}{
```

Comment:
This seems to be unfortunately cause new test failures because `master_user_options` is missing, @anGie44 may know more:

```
--- FAIL: TestAccAWSElasticSearchDomain_AdvancedSecurityOptions_UserDB (953.01s)
testing.go:684: Step 0 error: After applying this step, the plan was not empty:
DIFF:
UPDATE: aws_elasticsearch_domain.test
advanced_options.rest.action.multi.allow_explicit_index:                "true" => "true"
advanced_security_options.#:                                            "1" => "1"
advanced_security_options.0.enabled:                                    "true" => "true"
advanced_security_options.0.internal_user_database_enabled:             "true" => "true"
advanced_security_options.0.master_user_options.#:                      "0" => "1"
advanced_security_options.0.master_user_options.0.master_user_name:     "" => "testmasteruser"
advanced_security_options.0.master_user_options.0.master_user_password: "" => "Barbarbarbar1!"

--- FAIL: TestAccAWSElasticSearchDomain_AdvancedSecurityOptions_Disabled (965.41s)
testing.go:684: Step 0 error: After applying this step, the plan was not empty:
DIFF:
UPDATE: aws_elasticsearch_domain.test
advanced_options.rest.action.multi.allow_explicit_index:                "true" => "true"
advanced_security_options.#:                                            "1" => "1"
advanced_security_options.0.enabled:                                    "false" => "false"
advanced_security_options.0.internal_user_database_enabled:             "false" => "true"
advanced_security_options.0.master_user_options.#:                      "0" => "1"
advanced_security_options.0.master_user_options.0.master_user_name:     "" => "testmasteruser"
advanced_security_options.0.master_user_options.0.master_user_password: "" => "Barbarbarbar1!"

--- FAIL: TestAccAWSElasticSearchDomain_AdvancedSecurityOptions_IAM (1216.21s)
testing.go:684: Step 0 error: After applying this step, the plan was not empty:
DIFF:
UPDATE: aws_elasticsearch_domain.test
advanced_options.rest.action.multi.allow_explicit_index:           "true" => "true"
advanced_security_options.#:                                       "1" => "1"
advanced_security_options.0.enabled:                               "true" => "true"
advanced_security_options.0.internal_user_database_enabled:        "false" => "false"
advanced_security_options.0.master_user_options.#:                 "0" => "1"
advanced_security_options.0.master_user_options.0.master_user_arn: "" => "arn:aws:iam::*******:user/es-master-user-9000075685154087677"
```

Reply:
ahhh yep the explicit d.Set() were added here to workaround the entire block being overwritten. we wanted to keep the nested `master_user_options` block unchanged while also setting the `enabled` and `internal_user_database_enbaled` attributes which also exist in this outer `advanced_security_options` block

Reply:
it seems a bit unconventional but maybe for this case since we need to rebuild this `advanced_security_options` map when we're setting it back in state, we can explicitly get the `master_user_options` from state (as they're not returned from the API) with a d.Get() and append these attributes to the overall array being constructed in lines 717-720?

Reply:
@anGie44 yeah, that's the unfortunate way we've worked around these "missing nested configuration block in API response" situations in the past. Doesn't work on import for obvious reasons but there's only so much we can do. 😄

Reply:
@appilon @bflad https://github.com/terraform-providers/terraform-provider-aws/pull/14196 _should_ take care of these attributes

Reply:
no worries @anGie44 thanks for fixing those attrs! I still have more panics to settle before this PR is finally ready

Reply:
Let me know when your PR is done and I can review/merge it into this one. I think I got all of them now.